docs/en_US/HowToImplementTrainingService.md

```diff
@@ -0,0 +1,149 @@
+**How to Implement TrainingService in NNI**
+===
+
+## Overview
+TrainingService is a module related to platform management and job schedule in NNI. TrainingService is designed to be easily implemented, we define an abstract class TrainingService as the parent class of all kinds of TrainignService, users just need to  inherit the parent class and complete their own clild class if they want to implement customized TrainingService.
+
+## System architecture
+![](../img/NNIDesign.jpg)
+
+The brief system architecture of NNI is shown in the picture. NNIManager is the core management module of system, in charge of calling TrainingService to manage job and the communication between job and algorithm module. Dispatcher is a message processing center responsible for message dispatch. TrainingService is a module to manage trial jobs, it communicates with nniManager module, and has different instance according to different training platform. For the time being, NNI supports local platfrom, [remote server platfrom](RemoteMachineMode.md), [PAI platfrom](PAIMode.md), [kubeflow platform](KubeflowMode.md) and [FrameworkController platfrom](FrameworkController.md).   
+In this document, we introduce the brief design of TrainingService. If users want to add a new TrainingService instance, they just need to complete a child class to implement TrainingService, don't need to understand the code detail of NNIManager, Dispatcher or other modules.
+
+## Folder structure of code
+NNI's folder structure is shown below:
+```
+nni
+  |- deployment
+  |- docs
+  |- examaples
+  |- src
+  | |- nni_manager
+  | | |- common
+  | | |- config
+  | | |- core
+  | | |- coverage
+  | | |- dist
+  | | |- rest_server
+  | | |- training_service
+  | | | |- common
+  | | | |- kubernetes
+  | | | |- local
+  | | | |- pai
+  | | | |- remote_machine
+  | | | |- test
+  | |- sdk
+  | |- webui
+  |- test
+  |- tools
+  | |-nni_annotation
+  | |-nni_cmd
+  | |-nni_gpu_tool
+  | |-nni_trial_tool
+```
+`nni/src/` folder stores the most source code of system. The code in this folder is related to NNIManager, TrainingService, SDK, WebUI and other modules. Users could find the abstract class of TrainingService in `nni/src/nni_manager/common/trainingService.ts` file, and they should put their own implemented TrainingService in `nni/src/nni_manager/training_service` folder. If users have implemented their own TrainingService code, they should also supplement the unit test of the code, and place them in `nni/src/nni_manager/training_service/test` folder.
```

Comment:
"most source code of system":  suggest to replace "system" with NNI,  we do not say NNI as system.

Reply:
changed to NNI